### PR TITLE
fix: fullscreen frame scrolling bounce effect

### PR DESCRIPTION
**Pull request by xionglinlin**
Change the swipeview control to listview

Issue: https://github.com/linuxdeepin/developer-center/issues/10365

### CODE REVIEW

File: src/models/itemspagemodel.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    m_sourceModel = model;
+    emit sourceModelChanged(model);
+
+    m_topLevel = qobject_cast<ItemArrangementProxyModel*>(m_sourceModel)->itemsPage();
+    if (m_topLevel) {
+
+        m_topLevel->disconnect(SIGNAL(sigPageAdded(int, int)), this);
+        m_topLevel->disconnect(SIGNAL(sigPageRemoved(int, int)), this);
```
Comment (18202781743) [inline src/models/itemspagemodel.cpp:46]: 这个断开连接写在覆盖旧的值时,

File: src/models/itemspagemodel.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    QML_NAMED_ELEMENT(ItemsPageModel)
+
+public:
+    explicit ItemsPageModel(QObject *parent = nullptr);
+
+    Q_INVOKABLE int rowCount(const QModelIndex &parent = QModelIndex()) const override;
+    QVariant data(const QModelIndex &index, int role = Qt::DisplayRole) const override;
+
+    QAbstractItemModel *sourceModel() { return m_sourceModel; }
+    void setModel(QAbstractItemModel *model);
```
Comment (18202781743) [inline src/models/itemspagemodel.h:27]: setSourceModel, 与sourceModel保持一致

File: src/models/itemspagemodel.h [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+signals:
+    void sourceModelChanged(QObject *);
+
+private:
+    ItemsPage *m_topLevel;
```
Comment (18202781743) [inline src/models/itemspagemodel.h:33]: 直接访问原始的topLevel, 或者用QPointer包裹一下,以免被释放了访问野指针,

File: src/models/itemspagemodel.cpp [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+        connect(m_topLevel, &ItemsPage::sigPageAdded, this, [ = ] (int first, int last) {
+            beginInsertRows(QModelIndex(), first, last);
+            endInsertRows();
+        });
+
+        connect(m_topLevel, &ItemsPage::sigPageRemoved, this, [ = ] (int first, int last) {
```
Comment (18202781743) [inline src/models/itemspagemodel.cpp:53]: 只捕获this,

File: src/models/itemspagemodel.h (hunk 2 above)
Comment (xionglinlin) [reply src/models/itemspagemodel.h:27]: done

File: src/models/itemspagemodel.cpp (hunk 4 above)
Comment (xionglinlin) [reply src/models/itemspagemodel.cpp:53]: done

File: src/models/itemspagemodel.h (hunk 3 above)
Comment (xionglinlin) [reply src/models/itemspagemodel.h]: done

File: src/models/itemspagemodel.cpp (hunk 1 above)
Comment (xionglinlin) [reply src/models/itemspagemodel.cpp:48]: done

**Review by 18202781743 — APPROVED** [review]

**Pull request merged**